pkg/controller/v1beta1/inferenceservice/controller.go

```diff
@@ -223,7 +230,7 @@ func (r *InferenceServiceReconciler) updateStatus(desiredService *v1beta1api.Inf
 		// This is important because the copy we loaded from the informer's
 		// cache may be stale and we don't want to overwrite a prior update
 		// to status with this stale state.
-	} else if err := r.Status().Update(context.TODO(), desiredService); err != nil {
+	} else if err := r.Status().Patch(context.TODO(), desiredService, client.Merge, &client.PatchOptions{}); err != nil {
```

Comment:
Could add a comment explaining why we're doing a patch rather than update here?

Reply:
This is needed to prevent from overriding status when ModelMesh and KServe were not using the same InferenceService status spec. After ModelMesh changed to use the type definitions from KServe, Patch is no longer needed. Will change back to Update.